course-material/intro-to-sql/combining-data-from-multiple-tables.md

```diff
@@ -1,14 +1,305 @@
+
+To combine the `account`, `card`, `district`, and `link` tables, we can use the following SQL query:
+
+```{code-cell} ipython3
+%%sql 
+SELECT a.account_id, c.card_id, d.district_id, l.disp_id
+FROM s1.account AS a
+INNER JOIN s1.district AS d 
+ON a.district_id = d.district_id
+INNER JOIN s1.link AS l 
+ON l.account_id = a.account_id
+INNER JOIN s1.card AS c 
+ON c.disp_id = l.disp_id; 
+```
+
+In this query, we are performing a series of `INNER JOIN` operations to merge the tables based on the specified join conditions. The `ON` clause defines the relationship between the columns that are used for joining.
+
+By including the `*` in the `SELECT` statement, we are retrieving all columns from each of the joined tables in the result set. However, it is recommended to specify the specific columns you need to avoid unnecessary data retrieval.
+
+Executing this query will return a single table that combines the relevant columns from the `account`, `card`, `district`, and `link` tables.
+
+```tip
+The above query can also be written without the `INNER JOIN` clause! Another way to write the query is as follows:
+
+~~~python
+SELECT a.account_id, c.card_id, d.district_id, l.disp_id
+FROM s1.account as a, s1.card as c, s1.district as d, s1.link as l
+WHERE a.district_id = d.district_id AND
+      l.account_id = a.account_id AND
+      c.disp_id = l.disp_id;
+~~~
+```
+
+</details>
+<!-- #endregion -->
+
+### Question 2 (Medium):
+If tables `links` and `cards` did not have matching rows that are related to each other, what kind of join would you use to combine these tables? If you were to use this join, and then join the resulting table with the merged table of `account` and `district`, how would the final output differ from Question 1?
+
+<!-- #region -->
+<details>
+
+<summary>Answers</summary>
+
+A `FULL OUTER JOIN` would be used to combine the `links` and `cards` tables if we want to keep both the rows that can be matched and the unpaired rows. This is because, in this case, the tables `links` and `cards` do not have matching rows that are related to each other. Therefore, we need to account for the unpaired rows in the result set. 
+
+The `FULL OUTER JOIN` will return all rows from both tables, and if there are no matches, the columns from the other table will be filled with `NULL` values. This is in contrast to the `INNER JOIN`, which only returns rows that have matching values in both tables and, hence, no `NULL` values.
+
+Code to implement the `FULL OUTER JOIN` is shown below:
+
+```{code-cell} ipython3
+%%sql
+SELECT a.account_id, c.card_id, d.district_id, l.disp_id
+FROM s1.link AS l
+FULL OUTER JOIN s1.card AS c 
```

Comment:
I feel that questioning people about the full outer join is a bit unfair if we haven't shown them examples.

Reply:
shall include clear examples for it